stdlib/REPL/src/Terminals.jl

```diff
@@ -169,7 +171,101 @@ else
+
+    # Check COLORFGBG environment variable, which is set by rxvt and
+    # derivatives. This variable contains either two or three values
+    # separated by semicolons; we want the last value in either case.
+    # If this value is 0-6 or 8, our background is dark.
+    colorfgbg = get(ENV, "COLORFGBG", "")
+    sep = findlast(";", colorfgbg)
+    if sep !== nothing
+        # See e.g. <https://en.wikipedia.org/wiki/ANSI_escape_code>
+        # for the 16 possible colors
+        colorbg = tryparse(Int, colorfgbg[last(sep)+1:end])
+        if colorbg !== nothing && colorbg >= 0 && colorbg <= 15
+            return colorbg == 7 || colorbg >= 9
+        end
+    end
+
+    # Use OSC 11 escape sequence
+    BEL = "\a"                  # "\007"
+    ESC = "\e"                  # "\033"
+    OSC = "$(ESC)]"
+    send = "$(OSC)11;?$(BEL)"
+    reply = mktemp() do path, io
+        # Shell script idea taken from
+        # <https://stackoverflow.com/questions/2507337/is-there-a-way-to-determine-a-terminals-background-color>
+        cmds = ["printf %s '$send'",
+                "sleep 0.00000001", # xterm needs the sleep
+                "read -r recv",
+                "printf %s \"\$recv\" >'$path'"]
+        cmd = join(cmds, "; ")
+
+        oldstty = chomp(read(open(`stty -g`, "r", term.in_stream).out, String))
+        try
+            read(open(`stty raw -echo min 0 time 0`, "r", term.in_stream).out,
+                 String)
+            run(pipeline(`/bin/sh -c $cmd`;
+                         stdin=term.in_stream, stdout=term.out_stream))
+        finally
+            read(open(pipeline(`stty $oldstty`; stderr=devnull),
+                      "r", term.in_stream).out, String)
+        end
+        String(read(io))
+    end
+    # We expect a reply such as "$(OSC)11;rgb:00/00/00$(BEL)" or
+    # "$(OSC)11;rgba:0000/0000/0000/0000$(BEL)" or similar
+    if startswith(reply, "$(OSC)11;rgb")
+        col = findfirst(":", reply)
+        bel = findfirst(BEL, reply)
+        if col !== nothing && bel !== nothing
+            colors = reply[col.stop+1 : bel.start-1]
+            rgb = Float64[]
+            for str in split(colors, "/")
+                val = tryparse(UInt, str, base=16)
+                if val !== nothing
+                    maxval = UInt(1) << (4 * length(str)) - UInt(1)
+                    val /= maxval # normalize
+                    push!(rgb, val)
+                end
+            end
+            if length(rgb) >= 3
+                R, G, B = rgb
+                Y = 0.299 * R + 0.587 * G + 0.114 * B
+                return Y >= 0.5
+            end
+        end
+    end
+
+    # Guess from terminal name.
+    # This list of terminal names is taken from vim.
+    if term.term_type == "linux" ||         # Linux console
+        term.term_type == "screen.linux" || # Linux console with screen
+        startswith(term.term_type, "cygwin") || # Cygwin shell
+        startswith(term.term_type, "putty")     # Putty progrem
```

Comment:
I know this list comes from VIM, but it seems likely ancient. I think most terminals I use report as "xterm" now

Reply:
This is a list of terminals which default to using a dark background. The terminals which default to a light background (`xterm` and friends) are not in this list.